libsolutil/CommonData.h

```diff
@@ -163,6 +163,22 @@ auto applyMap(Container const& _c, Callable&& _op, OutputContainer _oc = OutputC
 	return _oc;
 }
 
+/// Filter a vector.
+/// Returns a copy of the vector after only taking indices `i` such that `_mask[i]` is true.
+template<typename T>
+std::vector<T> filter(std::vector<T> const& _vec, std::vector<bool> const& _mask)
```

Comment:
We could make this more generic by taking a callable instead of a mask.
In that case the caller interested in filtering-by-mask can just make a lambda out of that

Reply:
Ah, now I see this was just moved over. Fine to keep as is then.

Reply:
I only asked whether it makes sense moving it over, @ekpyron argued not as it may not be generic enough. I don't mind whichever way, perhaps @leonardoalt be a tie braker?

Reply:
> We could make this more generic by taking a callable instead of a mask.

I think it's more generic than just a `vector<T>` and a function `T -> Bool`, mainly because in this case the `vector<bool>` need not correspond to the input vector.